### DISCUSSION TITLE
请问 v2ray 内置的 DNS 能不能实现 DNS 解析结果改写的功能？

### DISCUSSION THREAD

**Discussion by jzp820927** [v2fly/v2ray-core#1130] [Q&A]
我想实现的功能，就是把所有使用了 cloudflare cdn 加速的域名，通过 v2ray 内部 DNS 解析的时候，DNS 响应结果全部固定返回为 1.1.1.1。原理大概是这样，先通过常规方式解析域名，然后判断解析结果中的 IP 地址，如果是 cloudflare 官方的 IP 地址段，就返回解析结果为 1.1.1.1。为什么需要这样的功能呢？因为我这边的宽带，针对 cloudflare 的地址段，有些能正常访问，有些又被阻断，在 cloudflare cdn 返回 IP 地址随机的情况下，就会造成有些网站能访问，有些不能，但是如果能全部返回一个固定的 cloudflare IP 的情况下（比如：1.1.1.1）那么就可以全部正常访问了。现在就想问下各位大佬 v2ray 的内置 DNS 能不能实现这个功能？如果不能，能否增加这个功能？如果不能增加这个功能，各位大佬能否告诉我下，有没有其他什么 DNS 软件，能实现这个功能的？

**Comment by kslr** [comment]
写一个简单的 dns 服务器

**Reply by jzp820927** [reply]
我对代码一点都不懂，能不能麻烦大佬帮忙写个？如果能合并到 v2fly 的 DNS 模块里面就更好了。

**Comment by rurirei** [comment]
write your own dns package.

**Reply by jzp820927** [reply]
我要是会写代码，就不会来发这个 discussions 咯。。。

**Comment by jzp820927** [comment]
能不能进来一位大佬，能给俺提供一点有价值的建议？
